app/src/main/java/com/duckduckgo/app/browser/ui/HttpAuthenticationDialogFragment.kt

```diff
@@ -60,18 +61,23 @@ class HttpAuthenticationDialogFragment : DialogFragment() {
                     request,
                     BasicAuthenticationCredentials(username = usernameInput.text.toString(), password = passwordInput.text.toString())
                 )
-
+                listener = null
             }.setNegativeButton(R.string.authenticationDialogNegativeButton) { _, _ ->
                 rootView.hideKeyboard()
                 listener?.cancelAuthentication(request)
+                listener = null
             }
             .setTitle(R.string.authenticationDialogTitle)
 
         val alert = alertBuilder.create()
         showKeyboard(usernameInput, alert)
-
         return alert
+    }
 
+    override fun onDismiss(dialog: DialogInterface) {
+        super.onDismiss(dialog)
+        //if listener is not null, user didn't press any button. We proceed to cancel.
```

Comment:
Can we make this self documenting? We could add a boolean method called `didUserCompleteAuthentication` or something similar and that would let us delete this comment.

Reply:
yeah, I didn't think about that. I like the idea of self documenting 💯